base/linalg/symmetric.jl

```diff
@@ -174,7 +174,54 @@ function svdvals!{T<:Real,S}(A::Union{Hermitian{T,S}, Symmetric{T,S}, Hermitian{
     return sort!(vals, rev = true)
 end
 
-#Matrix-valued functions
+#Matrix functions
+function ^(A::Symmetric, p::Integer)
+    if p < 0
+        return Symmetric(Base.power_by_squaring(inv(A), -p))
+    else
+        return Symmetric(Base.power_by_squaring(A, p))
+    end
+end
+function ^(A::Symmetric, p::Real)
+    F = eigfact(full(A))
```

Comment:
This seems problematic for complex-symmetric matrices, which may be nearly defective.